src/test/scala/chiselTests/LTLSpec.scala

```diff
@@ -241,16 +268,18 @@ class LTLSpec extends AnyFlatSpec with Matchers {
     chirrtl should include("connect verif.property, ltl_clock.out")
 
     // with disable; emitted as `assert(disable(a, b))`
+    chirrtl should include("node x2 = bits(b, 0, 0)")
```

Comment:
I have no idea where this x2 comes from and at this point I'm afraid to ask

Jokes aside, I'm looking into it.

Reply:
Looks like a product of the desugaring of the default API:
```scala
class Example extends RawModule {
  val a, b = IO(Input(Bool()))
  val c = IO(Input(Clock()))
  AssertProperty(a, disable = Some(b.asDisable))
}
```

By the time the Chisel plugin sees it, it has be lowered to:

```scala
class Example extends chisel3.RawModule {
  def <init>(): Example = {
    Example.super.<init>();
    ()
  };
  private[this] val a: chisel3.Bool = Example.this.IO[chisel3.Bool](chisel3.Input.apply[chisel3.Bool](chisel3.`package`.Bool.apply()))((chisel3.experimental.SourceLine.apply("src/main/scala/Jack.scala", 9, 16): chisel3.experimental.SourceInfo));
  <stable> <accessor> def a: chisel3.Bool = Example.this.a;
  private[this] val b: chisel3.Bool = Example.this.IO[chisel3.Bool](chisel3.Input.apply[chisel3.Bool](chisel3.`package`.Bool.apply()))((chisel3.experimental.SourceLine.apply("src/main/scala/Jack.scala", 9, 16): chisel3.experimental.SourceInfo));
  <stable> <accessor> def b: chisel3.Bool = Example.this.b;
  private[this] val c: chisel3.Clock = Example.this.IO[chisel3.Clock](chisel3.Input.apply[chisel3.Clock](chisel3.Clock.apply()))((chisel3.experimental.SourceLine.apply("src/main/scala/Jack.scala", 10, 13): chisel3.experimental.SourceInfo));
  <stable> <accessor> def c: chisel3.Clock = Example.this.c;
  {
    <artifact> val x$1: chisel3.ltl.Sequence.BoolSequence = chisel3.ltl.Sequence.BoolSequence(Example.this.a);
    <artifact> val x$2: Some[chisel3.Disable] @scala.reflect.internal.annotations.uncheckedBounds = scala.Some.apply[chisel3.Disable](Example.this.b.do_asDisable(scala.Predef.implicitly[chisel3.experimental.SourceInfo]((chisel3.experimental.SourceLine.apply("src/main/scala/Jack.scala", 12, 38): chisel3.experimental.SourceInfo))));
    <artifact> val x$3: Option[chisel3.Clock] @scala.reflect.internal.annotations.uncheckedBounds = chisel3.ltl.AssertProperty.apply$default$2;
    <artifact> val x$4: Option[String] @scala.reflect.internal.annotations.uncheckedBounds = chisel3.ltl.AssertProperty.apply$default$4;
    chisel3.ltl.AssertProperty.apply(x$1, x$3, x$2, x$4)
  }
}
```

I think there's an interesting question about having the plugin detect these synthetic names (I think that's what the `<artifact>` decorator is indicating) and ignore them. That could have wide-ranging impacts on emitted FIRRTL and thus Verilog though so I think it should be done completely separately from this PR.

Reply:
If you change the input Chisel to:
```scala
class Example extends RawModule {
  val a, b = IO(Input(Bool()))
  val c = IO(Input(Clock()))
  AssertProperty(a, None, Some(b.asDisable), None)
}
```

Then these artifact vals don't exist and the name of the node is `_T`.

FWIW I'd like to get rid of the need for this node (the pointless bit-extract on an already 1-bit UInt), but that also is a separate PR.

Reply:
This is surprising.

This is the type of thing that FileCheck would help with. (The test really shouldn't care about the change even if we should be tracking this with an improvement to the naming plugin.)

Reply:
This solved me a long time question...
Thanks Jack...
You might need to take a look at your diplomacy framework, IIRC, `BundleBridge` has the same issue, and I didn't spend time on it, but directly suggest name,